server/src/main/java/de/uftos/repositories/solver/timefold/constraints/ConstraintDefinitionFactory.java

```diff
@@ -0,0 +1,1073 @@
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.IS_EMPTY) {
+      throw new IllegalStateException();
+    }
+    OperatorDto isEmpty = (OperatorDto) ast;
+    if (isEmpty.parameters().size() != 1) {
+      throw new IllegalStateException();
+    }
+    Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> set =
+        convertSet(isEmpty.parameters().getFirst(), params);
+    return (parameters) -> set.apply(parameters).isEmpty();
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Number> convertSize(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.SIZE) {
+      throw new IllegalStateException();
+    }
+    OperatorDto size = (OperatorDto) ast;
+    if (size.parameters().size() != 1) {
+      throw new IllegalStateException();
+    }
+    Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> set =
+        convertSet(size.parameters().getFirst(), params);
+    return (parameters) -> new Number(set.apply(parameters).size());
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> convertSet(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    return switch (ast.getToken()) {
+      case RESOURCE_SET -> convertResourceSet(ast, params);
+      case NUMBER_SET -> convertNumberSet(ast, params);
+      default -> throw new IllegalStateException();
+    };
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+      convertResourceSet(AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.RESOURCE_SET) {
+      throw new IllegalStateException();
+    }
+    SetDto set = (SetDto) ast;
+
+    Function<List<ResourceTimefoldInstance>, ResourceTimefoldInstance> name;
+
+    if (set.setName().getToken() == UcdlToken.VALUE_REFERENCE) {
+      name = convertValueReference(set.setName(), params);
+    } else {
+      name = convertElement(set.setName(), params);
+    }
+
+    List<Function<
+        List<ResourceTimefoldInstance>,
+        Function<Set<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+        >> modifiers = new ArrayList<>();
+
+    for (AbstractSyntaxTreeDto modifier : set.modifiers()) {
+      if (modifier.getToken() == UcdlToken.ATTRIBUTE) {
+        modifiers.add(convertAttribute(modifier, params));
+      } else {
+        modifiers.add(convertFilter(modifier, params));
+      }
+    }
+
+    return (parameters) -> {
+      Set<ResourceTimefoldInstance> setReference = new HashSet<>();
+      setReference.add(name.apply(parameters));
+
+      for (Function<
+            List<ResourceTimefoldInstance>,
+            Function<Set<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+            > function : modifiers) {
```

Comment:
Honestly, you should just use `var` here. Java doesn't allow you to specify type aliases because it's stupid so I think `var` is the best solution

Reply:
I have no idea, what `var` is supposed to do and i don`t undestand, what you want to be changed here.

Reply:
You can use `var` instead of explicitly defining a type. So for instance these two examples are exactly the same:
```java
var foo = "Hello World!"; // type String
String bar = "Hello World!"; // type String
```

Reply:
So do I understand it correctly, that you want me to replace `List<...>` with `var`?

Reply:
Yes!

Reply:
I honestly don't like that.
I prefer ugly but visible types instead of just writing `var` and leaving the rest to the compiler.